samtranslator/model/role_utils/role_constructor.py

```diff
@@ -83,8 +134,12 @@ def construct_role_for_resource(  # type: ignore[no-untyped-def] # noqa: too-man
             #
 
             policy_arn = policy_entry.data
-            if isinstance(policy_entry.data, str) and policy_entry.data in managed_policy_map:
-                policy_arn = managed_policy_map[policy_entry.data]
+            if isinstance(policy_arn, str):
```

Comment:
For my own curiosity, what should happen if this is the empty string?

Reply:
It falls back to whatever the string is, so empty string. And deployment will fail.